vllm/executor/gpu_executor.py

```diff
@@ -11,6 +11,15 @@
 logger = init_logger(__name__)
 
 
+def create_worker(worker_module_name, worker_class_name, **kwargs):
```

Comment:
The changes in this file are to ensure that the function/args passed to the new worker process don't contain a reference to the executor instance itself (`self`).

Reply:
What happens if we don't have this? Is this more of a code cleanliness thing?

Reply:
@andoorve no, for some reason there are pickling errors when forking the workers without this. What's confusing is why it works in the non-PP case.

Reply:
how does this solve the problem?

Reply:
okay, I think the problem is that we use `partial` later:

```python
partial(
                        self._create_worker,
                        rank=rank,
                        local_rank=rank,
                        distributed_init_method=distributed_init_method,
                    ))
```

Reply:
@youkaichao yes and in particular the implicit reference to `self` held by the instance method it contains.